### PR TITLE
Tests do not allow to use SEND_OUT if the chosen mon is fainted

### LINKED ISSUE #3747: Tests allow to use SEND_OUT if the chosen mon is fainted
### Description

I was working in an Supreme Overlord test and found that `SEND_OUT` was abled sent in a mon with 0 HP. I would assume that it should be set as `INVALID`.
mGBA_hJ4q6FT1Eg
```
DOUBLE_BATTLE_TEST("Supreme Overlord's boost caps at 1.5x")
{
    //bool32 faintCount = 0;
    //PARAMETRIZE { faintCount = 5; }
    //PARAMETRIZE { faintCount = 6; }
    GIVEN {
        PLAYER(SPECIES_PAWNIARD);
        PLAYER(SPECIES_PAWNIARD);
        PLAYER(SPECIES_PAWNIARD);
        PLAYER(SPECIES_KINGAMBIT) { Ability(ABILITY_SUPREME_OVERLORD); }
        OPPONENT(SPECIES_WOBBUFFET);
        OPPONENT(SPECIES_WOBBUFFET);
    } WHEN {
        TURN { MOVE(playerLeft, MOVE_MEMENTO, target: opponentRight); SEND_OUT(playerLeft, 2); }
        TURN { MOVE(playerLeft, MOVE_MEMENTO, target: opponentRight); SEND_OUT(playerLeft, 2); USE_ITEM(playerRight, ITEM_REVIVE, 0); }
        TURN { MOVE(playerLeft, MOVE_MEMENTO, target: opponentRight); SEND_OUT(playerLeft, 2); USE_ITEM(playerRight, ITEM_REVIVE, 0); }
        //TURN { MOVE(playerLeft, MOVE_TACKLE, target: opponentLeft); }
    } SCENE {
        //ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, playerLeft);
        //HP_BAR(opponentLeft, captureDamage: &results[i].damage);
    //} FINALLY {
    //    EXPECT_MUL_EQ(results[0].damage, Q_4_12(1.1), results[1].damage);
    }
}
```

### Version

upcoming (Edge)

### Upcoming Version

fbbee3988680730fc1b5ff25637deb7b81aabd53

### Discord contact info

AsparagusEduardo

### PR DESCRIPTION
**Pull request by DizzyEggg**
If the sent out pokemon has 0 hp, the tests fails as invalid.

Fixes #3747

Note: Squash when merging.

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -6157,6 +6157,21 @@ static void Cmd_switchindataupdate(void)
...
 
+    // Edge case: the sent out pokemon has 0 HP. This should never happen.
+    if (gBattleMons[battler].hp == 0)
+    {
+        // If it's a test, mark it as invalid.
+        if (gTestRunnerEnabled)
+        {
+            TestRunner_Battle_InvalidNoHPMon(battler, gBattlerPartyIndexes[battler]);
+        }
+        // TODO: Handle in-game scenario.
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:6168]: Do you have an idea of what should happen in this situation in-game?
If not, I think combining the gTestRunnerEnabled and HP checks should suffice.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: src/battle_script_commands.c (hunk 1 above)
Comment (DizzyEggg) [reply src/battle_script_commands.c]: I made it so the game looks for a valid replacement. Though, this should never happen anyway(unless cheats or bugs).

**Review by Bassoonian — APPROVED** [review]
